src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -633,8 +639,7 @@ static int parse_request(uint8_t code, uint8_t len, const void *option, void *us
                 break;
         case SD_DHCP_OPTION_MAXIMUM_MESSAGE_SIZE:
                 if (len == 2)
-                        req->max_optlen = be16toh(*(be16_t*)option) -
-                                          - sizeof(DHCPPacket);
+                        req->max_optlen = unaligned_read_be16(option) + sizeof(DHCPPacket);
```

Comment:
Yikes. Indeed. This is a bug. That should have been one `-`, not two.

For the record: the option indicates the maximum size of the DHCP packet, we store instead the maximum size of the options portion of the packet (which is the variable part). Due to this bug it would have been possible for a server to assume a client could handle larger packets than they could and hence send over-large packets, which would have been dropped by the client.